core/orchestrator.go

```diff
@@ -31,6 +31,11 @@ import (
 
 var transcodeLoopTimeout = 1 * time.Minute
 
+// Gives us more control of "timeout" / cancellation behavior during testing
+var transcodeLoopContext = func() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), transcodeLoopTimeout)
```

Comment:
Would be nice if we can set the transcode loop timeout based on the length of the segment

Reply:
Perhaps - is there a particular case that would be helped by doing it like this? Or a particular formulation you were thinking of?

Some additional notes around depending on the segment length (duration) in the orchestrator.

* Orchestrator currently doesn't know the length (duration) of the segment that it receives. We can add some [signaling](https://github.com/livepeer/go-livepeer/blob/909376fe6ee07144b0bed94f5adead98518d6697/net/lp_rpc.proto#L104-L126) for that data, but I don't believe we can outright trust that bit of data from the B.

* This timer only begins counting *after* a segment completes processing, so there's no possibility of cancelling the loop while a segment is in-flight. As soon as a new segment comes in, the timer is cancelled, and the segment itself can potentially take a very long time to process [1].

[1] If this sounds like a problem: it is! But that's not for this PR to solve.

* The default timeout is 1 minute, which should be plenty for live segmented streams, where we try to segment on 2-second boundaries. Other types of streams (eg, VOD) are not under consideration here, and the workflow for that would be quite different.

* If the transcode loop timeout is dependent on the segment length, setting a hard cap upper limit is still probably desirable to preserve O resources.